python/tchannel/tornado/tornado_req_handler.py

```diff
@@ -0,0 +1,92 @@
+import json
+from tornado import httputil
+from tornado.httputil import RequestStartLine
+from ..req_handler import RequestHandler
+
+
+class TornadoRequestHandler(RequestHandler):
+    def __init__(self, app):
+        self.request_callback = app
+
+    def start_serving(self, request_conn):
+        return _ServerRequestAdapter(self, request_conn)
+
+    def handle_request(self, context, conn):
+        """Handle incoming request
+
+        :param context: incoming message context
+        :param conn: incoming connection
+        """
+        request_delegate = self.start_serving(conn)
+        message = context.message
+        # process http message
+        if message.headers["as"] == "http":
+            method = "GET"
+            if (hasattr(message, "arg_3") and
+                    message.arg_3 is not None and
+                    message.arg_3 != ""):
+                method = "POST"
+
+            start_line = RequestStartLine(method, message.arg_1, 'HTTP/1.1')
+            try:
+                headers = json.loads(message.arg_2)
+            except:
+                headers = {}
```

Comment:
In case of `http`, `arg_2` contains headers as a JSON array, not a dict. Make this an empty list.